File: TPoint/TPoint/Domain/Entities/Team.swift
//
//  Team.swift
//  TPoint
//

import Foundation

enum TeamType: String, CaseIterable {
    case teamA = "A"
    case teamB = "B"

    var displayName: String {
        switch self {
        case .teamA: return L10n.teamA
        case .teamB: return L10n.teamB
        }
    }

    var shortName: String {
        rawValue
    }

    var opponent: TeamType {
        switch self {
        case .teamA: return .teamB
        case .teamB: return .teamA
        }
    }
}

struct Team {
    let type: TeamType
    var totalScore: Int = 0

    var displayName: String {
        type.displayName
    }
}

File: TichuTallyAndroid/app/src/main/java/com/tichutally/app/domain/model/Team.kt
package com.tichutally.app.domain.model

enum class TeamType(val shortName: String) {
    TEAM_A("A"),
    TEAM_B("B");

    val opponent: TeamType
        get() = when (this) {
            TEAM_A -> TEAM_B
            TEAM_B -> TEAM_A
        }
}

data class Team(
    val type: TeamType,
    val totalScore: Int = 0
)
